include/xrpl/ledger/View.h

```diff
@@ -875,6 +1094,26 @@ canTransfer(
     AccountID const& from,
     AccountID const& to);
 
+[[nodiscard]] TER
+canTransfer(
+    ReadView const& view,
+    Issue const& issue,
+    AccountID const& from,
+    AccountID const& to);
+
+[[nodiscard]] TER inline canTransfer(
+    ReadView const& view,
+    Asset const& asset,
+    AccountID const& from,
+    AccountID const& to)
+{
+    return std::visit(
+        [&]<ValidIssueType TIss>(TIss const& issue) -> TER {
+            return canTransfer(view, issue, from, to);
+        },
+        asset.value());
+}
+
```

Comment:
this would be a good place to check for depositPreauth/credentials as well

Reply:
> this would be a good place to check for depositPreauth/credentials as well

Probably. I'll have to follow up on this, if there isn't already a pending ticket being worked.